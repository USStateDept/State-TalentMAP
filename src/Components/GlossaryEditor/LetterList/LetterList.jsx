import PropTypes from 'prop-types';
import InteractiveElement from '../../InteractiveElement';

const LetterList = ({ letters, onClick }) => (
  <div className="letters-list">
    {
      letters.map(letter =>
        (
          <InteractiveElement
            type="span"
            key={letter}
            role="link"
            title={`View terms that begin with ${letter}`}
            className="letter-link"
            onClick={() => onClick(letter)}
          >
            {letter}
          </InteractiveElement>
        ),
      )
    }
    <InteractiveElement
      type="span"
      role="link"
      title="View all terms"
      className="letter-link letter-link--all"
      onClick={() => onClick(null)}
    >
      All
    </InteractiveElement>
  </div>
);

LetterList.propTypes = {
  letters: PropTypes.arrayOf(PropTypes.string).isRequired,
  onClick: PropTypes.func.isRequired,
};

export default LetterList;
