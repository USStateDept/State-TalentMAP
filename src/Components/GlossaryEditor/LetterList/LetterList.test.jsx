import sinon from 'sinon';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import LetterList from './LetterList';

describe('LetterListComponent', () => {
  const props = {
    letters: ['#', 'A', 'B'],
    onClick: () => {},
  };
  it('is defined', () => {
    const wrapper = shallow(<LetterList
      {...props}
    />);
    expect(wrapper).toBeDefined();
  });

  it('can click on a letter link', () => {
    const spy = sinon.spy();
    const wrapper = shallow(<LetterList
      {...props}
      onClick={spy}
    />);
    expect(wrapper.find('InteractiveElement').first().simulate('click'));
    sinon.assert.calledOnce(spy);
  });

  it('can click on the All link', () => {
    const spy = sinon.spy();
    const wrapper = shallow(<LetterList
      {...props}
      onClick={spy}
    />);
    expect(wrapper.find('.letter-link--all').simulate('click'));
    sinon.assert.calledOnce(spy);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<LetterList
      {...props}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
