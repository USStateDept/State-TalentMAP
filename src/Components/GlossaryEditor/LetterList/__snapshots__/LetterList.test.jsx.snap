// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterListComponent matches snapshot 1`] = `
<div
  className="letters-list"
>
  <InteractiveElement
    className="letter-link"
    key="#"
    onClick={[Function]}
    role="link"
    title="View terms that begin with #"
    type="span"
  >
    #
  </InteractiveElement>
  <InteractiveElement
    className="letter-link"
    key="A"
    onClick={[Function]}
    role="link"
    title="View terms that begin with A"
    type="span"
  >
    A
  </InteractiveElement>
  <InteractiveElement
    className="letter-link"
    key="B"
    onClick={[Function]}
    role="link"
    title="View terms that begin with B"
    type="span"
  >
    B
  </InteractiveElement>
  <InteractiveElement
    className="letter-link letter-link--all"
    onClick={[Function]}
    role="link"
    title="View all terms"
    type="span"
  >
    All
  </InteractiveElement>
</div>
`;
